common/djangoapps/pipeline_mako/templates/static_content.html

```diff
@@ -154,6 +154,35 @@
     </script>
 </%def>
 
+<%def name="require_page(page_name, class_name)">
+    <%doc>
+      Loads Javascript onto your page synchronously.
+      Uses RequireJS in development and a plain script tag in production.
+      Use this form of require_module for all new code.
+    </%doc>
+    <%
+        body = capture(caller.body)
+    %>
+    <script type="text/javascript">
+        if (typeof pageFactoryArguments == "undefined") {
+            var pageFactoryArguments = {};
+        }
+        % if body:
+        pageFactoryArguments["${class_name}"] = [${ body | n, decode.utf8 }];
```

Comment:
:+1: on using `class_name` as the key here, but as we go we should keep an eye out for pages with multiple invocations of the same factory.